### PR TITLE
feat: add sidebar

### PR DESCRIPTION
**Pull request by satazor**
See: https://github.com/ipfs-shipyard/pm-idm/issues/120

### CODE REVIEW

File: src/shell/sidebar/Sidebar.js [hunk 1]
```diff
@@ -1,28 +1,124 @@
...
 import styles from './Sidebar.css';
 
-class Sidebar extends Component {
+const OPEN_DELAY_DURATION = 250;
+const OPEN_TRANSITION_DURATION = 300;
+const OPEN_TRANSITION_CLASSNAMES = { enter: styles.open, enterDone: styles.open, exit: styles.close };
+const STAGGER_FIXED_TRANSITION_DELAY = 50;
+const STAGGER_TRANSITION_DELAY = 30;
+
+const mockIdentities = [
```
Comment (satazor) [inline src/shell/sidebar/Sidebar.js:19]: This will be removed once `idmWallet.identities.list` becomes available

File: src/Boot.css [hunk 2]
```diff
@@ -1,7 +1,13 @@
...
-    display: inline-flex;
-    flex-direction: column;
-    justify-content: center;
-    align-content: center;
+    display: flex;
+    flex: 1 0 auto;
+
+    & .loading,
+    & .error {
+        margin: auto auto;
```
Comment (PedroMiguelSS) [inline src/Boot.css]: Isn't this the same as `margin: auto`?

File: src/shell/sidebar/Sidebar.css [hunk 3]
```diff
@@ -1,3 +1,127 @@
...
+            height: var(--bottom-height);
+            margin: 0;
+            padding: 1.2rem 0;
+            flex: 0 0 auto;
+            list-style: none;
+        }
+    }
+
+    /* Safe area
+       Prevents someone interacting with the scrollbar to accidently close the sidebar
```
Comment (PedroMiguelSS) [inline src/shell/sidebar/Sidebar.css]: Can we change this comment to something like: 

> Outside area from the sidebar that prevents it from closing on accidental mouse leave.
> As an example: while scrolling fast, the mouse might leave by mistake.

File: src/shell/sidebar/Sidebar.js [hunk 4]
```diff
@@ -1,28 +1,124 @@
...
+        this.setState({ open: false });
+    }
+
+    open() {
+        clearTimeout(this.enterTimeout);
+        this.enterTimeout = setTimeout(() => this.setState({ open: true }), OPEN_DELAY_DURATION);
+    }
+
+    // We use onMouseOver instead of onMouseEnter because the user might click the avatar/links
+    // while still in the sidebar. This way, the sidebar will open again as soon as anotherelement
```
Comment (PedroMiguelSS) [inline src/shell/sidebar/Sidebar.js]: Change `anotherelement` to `another element`.

**Review by dominguesgm — APPROVED** [review]

**Pull request merged**